dotnet/src/Connectors/Connectors.MistralAI/MistralAIKernelBuilderExtensions.cs

```diff
@@ -0,0 +1,120 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Net.Http;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.SemanticKernel.ChatCompletion;
+using Microsoft.SemanticKernel.Embeddings;
+using Microsoft.SemanticKernel.Http;
+
+namespace Microsoft.SemanticKernel.Connectors.MistralAI;
+
+/// <summary>
+/// Provides extension methods for the <see cref="IKernelBuilder"/> class to configure Mistral connectors.
+/// </summary>
+public static class MistralAIKernelBuilderExtensions
+{
+    /// <summary>
+    /// Adds an Mistral chat completion service with the specified configuration.
+    /// </summary>
+    /// <param name="builder">The <see cref="IKernelBuilder"/> instance to augment.</param>
+    /// <param name="model">The name of the Mistral model.</param>
+    /// <param name="endpoint">The endpoint URL for the chat completion service.</param>
+    /// <param name="apiKey">The API key required for accessing the Mistral service.</param>
+    /// <param name="serviceId">A local identifier for the given AI service.</param>
+    /// <returns>The same instance as <paramref name="services"/>.</returns>
+    public static IKernelBuilder AddMistralChatCompletion(
+        this IKernelBuilder builder,
+        string model,
+        Uri? endpoint = null,
+        string? apiKey = null,
+        string? serviceId = null)
```

Comment:
What is purpose of this overload since you have overload with optional httpClient?